src/librustc/mir/interpret/mod.rs

```diff
@@ -166,9 +166,15 @@ pub enum LitToConstError {
     Reported,
 }
 
-#[derive(Copy, Clone, Eq, Hash, Ord, PartialEq, PartialOrd, Debug)]
+#[derive(Copy, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct AllocId(pub u64);
```

Comment:
A bit surprised this is not using the newtyped index stuff.

Reply:
I didn't do that because that's hardcoded to `u32`. I'll adjust the macro to support changing the underlying type and make `AllocId` a `newtype_index`.

It's not quite as helpful as with other index vecs, as we can't use `IndexVec` because the ids aren't consecutive.

Reply:
Why not use `u32` though?
> we can't use `IndexVec` because the ids aren't consecutive

Ah nevermind.